nautobot/project-static/js/forms.js

```diff
@@ -160,6 +160,53 @@ function initializeColorPicker(context, dropdownParent=null){
     });
 }
 
+/**
+ * Retrieves the value of a property from a nested object using a string path.
+ *
+ * This method supports accessing deeply nested properties within an object.
+ * It is created to support extraction of nested values in the display-field
+ * and value-field for DynamicChoiceField.
+ *
+ * @param {Object} response - The object from which to retrieve the value.
+ * @param {string} fieldPath - The string representing the path to the desired property.
+ * @returns {*} The value of the specified property, or null if the path is invalid or the object is not found.
+ *
+ * @example
+ * let response = {
+ *   "id": 1234,
+ *   "vlan": {
+ *     "name": "myvlan"
+ *   },
+ *   "interfaces": [
+ *     { "name": "eth0", "status": "up" },
+ *     { "name": "eth1", "status": "down" }
+ *   ]
+ * }
+ * // returns "myvlan"
+ * resolvePath(response, "vlan.name")
+ *
+ * // returns "eth0"
+ * resolvePath(response, "interfaces[0].name")
+ *
+ * // returns "eth0"
+ * resolvePath(response, "interfaces.0.name")
+ */
+function resolvePath(response, fieldPath) {
+    if (!fieldPath)
+        return null;
+
+    if (typeof response !== 'object' || response === null || !response) {
+        console.error('Invalid response object');
+        return null;
+    }
+
+    return fieldPath
+           .replace(/\[|\]\.?/g, '.')
+           .split('.')
+           .filter(value => value)
```

Comment:
What cases is this `.filter()` guarding against? Is it needed?

Reply:
It removes 'empty'  fields, for example, if you said `interface..name` it would still work rather than silently failing.